app/models/facility.rb

```diff
@@ -60,7 +60,7 @@ def self.cross_facility
 
   def products(type = nil)
     if type
-      super.where(type: type)
+      self.public_send(type.to_s.pluralize.underscore.to_sym)
```

Comment:
Rails 5 really didn't like the previous way this method was structured. This refactoring is a little clunky--let me know if you have a better idea.

Reply:
I'll need to think about this. I think I ran into something similar with an association pointing at an STI table on another project. I need to remember how I handled that. Might also be worth looking into how this method is used.

Reply:
Now that I see what this is doing, it's definitely something hacky. I would vote for killing this method off and instead replacing usages of it with a new scope on Product. 

```ruby
# current
current_facility.products("Instrument").find_by!(url_name: params[id_param])

# proposed
current_facility.products.of_type("Instrument").find_by!(url_name: params[id_param])
```